migration.rst

```diff
@@ -0,0 +1,412 @@
+making sure that external systems do not change your production environment,
+e.g. provide a separate test database with (anonymized) data from a production
+system or being able to setup a new schema with a basic dataset for your test
+environment. Since these tests do not rely as much on isolating testable code
+and instead look at the interconnected system, writing them is usually easier
+and more productive when doing a migration. You can then limit your effort on
+writing lower level tests on parts of the code that you have to change or
+replace in the new application making sure it is testable right from the start.
+
+There are tools aimed at End-to-End testing you can use such as
+`Symfony Panther`_ or you can write :ref:`functional_tests` in the new Symfony
+application as soon as the initial setup is completed. For example you can add
+so called Smoke Tests, which only ensure a certain path is accessible by
+checking the HTTP status code returned or looking for a text snippet from the
+page.
+
+Introducing Symfony to the Existing Application
+-----------------------------------------------
+
+The following instructions only provide an outline of common tasks for
+setting up a Symfony application that falls back to a legacy application
+whenever a route is not accessible. Your mileage may vary and likely you
+will need to adjust some of this or even provide additional configuration
+or retrofitting to make it work with your application. This guide is not
+supposed to be comprehensive and instead aims to be a starting point.
+
+.. tip::
+
+    If you get stuck or need additional help you can reach out to the
+    :doc:`Symfony community<contributing/community>` whenever you need
+    concrete feedback on an issue you are facing.
+
+When looking at how a typical PHP application is bootstrapped there are
+two major approaches. Nowadays most frameworks provide a so called
+front controller which acts as an entrypoint. No matter which URL-path
+in your application you are going to, every request is being sent to
+this front controller, which then determines which parts of your
+application to load, e.g. which controller and action to call. This is
+also the approach that Symfony takes with ``public/index.php`` being
+the front controller. Especially in older applications it was common
+that different paths were handled by different PHP files.
+
+In any case you have to create a ``public/index.php`` that will start
+your Symfony application by either copying the file from the
+`FrameworkBundle`_-recipe or by using Flex and requiring the
+FrameworkBundle. You will also likely have to update you web server
+(e.g. Apache or nginx) to always use this front controller. You can
+look at :doc:`Web Server Configuration <setup/web_server_configuration>`
+for examples on how this might look. For example when using Apache you can
+use Rewrite Rules to ensure PHP files are ignored and instead only index.php
+is called::
+
+    RewriteEngine On
```

Comment:
Pleas use „.. code-block:: bash“ or „text“

@javiereguilaz WDYT?

Reply:
I changed it to `.. code-block:: text` for now

Reply:
We can use `.. code-block:: apache`  as in other parts of the docs where we show Apache config.